backend/fms_core/template_importer/row_handlers/normalization_planning/normalization_planning.py

```diff
@@ -0,0 +1,128 @@
+from fms_core.template_importer.row_handlers._generic import GenericRowHandler
+from fms_core.template_importer._constants import VALID_NORM_CHOICES
+
+from fms_core.services.container import get_container, is_container_valid
+from fms_core.services.sample import get_sample_from_container
+
+from fms_core.utils import convert_concentration_from_nm_to_ngbyul, decimal_rounded_to_precision
+
+import decimal
+
+class NormalizationPlanningRowHandler(GenericRowHandler):
+    """
+         Extracts the information of each row in a template sheet and validates it.
+
+         Returns:
+             The errors and warnings of the row in question after validation.
+    """
+
+    def process_row_inner(self, source_sample, destination_sample, measurements, robot):
+        concentration_nguL = None
+        concentration_nm = None
+        combined_concentration_nguL = None
+
+        # Check if robot output choice is valid
+        if robot["norm_choice"] not in VALID_NORM_CHOICES:
+            self.errors['robot_norm_choice'] = f"Robot normalization choice must be chosen among the following choices : {VALID_NORM_CHOICES}."
+
+        # Check case when none of the options were provided
+        if all([measurements['concentration_nm'] is None,
+                measurements['concentration_ngul'] is None,
+                measurements['na_quantity'] is None]):
+            self.errors['concentration'] = 'One option (A, B or C) should be specified.'
+
+        # Check that there's only one option provided
+        if sum([measurements['concentration_nm'] is not None,
+                measurements['concentration_ngul'] is not None,
+                measurements['na_quantity'] is not None]) != 1:
+            self.errors['concentration'] = 'Only one option must be specified out  of the following: NA quantity, conc. ng/uL or conc. nM.'
+
+        source_sample_obj, self.errors['sample'], self.warnings['sample'] = get_sample_from_container(
+            barcode=source_sample['container']['barcode'],
+            coordinates=source_sample['coordinates'])
+
+        if source_sample_obj.concentration is None:
```

Comment:
We might wanna ad an `if source_sample_obj: `and then continue after line 44. If source sample is None then this will crash.